src/main/java/edu/cornell/kfs/module/purap/service/impl/CuElectronicInvoiceHelperServiceImpl.java

```diff
@@ -1666,4 +1686,92 @@ protected void addRejectReasonsToNote(String rejectReasons, ElectronicInvoiceRej
+            LOG.error("addNamespaceDefinition() Error getting document builder - " + e.getMessage());
+            throw new RuntimeException(e);
+        }
+
+        Document xmlDoc = null;
+
+        try {
+            LOG.info("addNamespaceDefinition: builder.parse");
+            xmlDoc = builder.parse(invoiceFile);
+        } catch (Exception e) {
+            LOG.info("addNamespaceDefinition: Error parsing the file - " + e.getMessage());
+            rejectElectronicInvoiceFile(eInvoiceLoad, UNKNOWN_DUNS_IDENTIFIER, invoiceFile, e.getMessage(), PurapConstants.ElectronicInvoice.FILE_FORMAT_INVALID);
+            return null;
+        }
+        LOG.info("addNamespaceDefinition: xmlDoc.getDocumentElement()");
+
+        Node node = xmlDoc.getDocumentElement();
+        Element element = (Element) node;
+
+        removeUnsupportedXmlnsAttributes(element);
+
+        String xmlnsValue = element.getAttribute("xmlns");
+        String xmlnsXsiValue = element.getAttribute("xmlns:xsi");
+
+        LOG.info("addNamespaceDefinition: getInvoiceFile");
+
+        if (StringUtils.equals(xmlnsValue, "http://www.kuali.org/kfs/purap/electronicInvoice") &&
+            StringUtils.equals(xmlnsXsiValue, "http://www.w3.org/2001/XMLSchema-instance")) {
+            LOG.info("addNamespaceDefinition: xmlns and xmlns:xsi attributes already exists in the invoice xml");
+        } else {
+            element.setAttribute("xmlns", "http://www.kuali.org/kfs/purap/electronicInvoice");
+            element.setAttribute("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance");
+        }
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try {
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            Source source = new DOMSource(xmlDoc);
+            Result result = new StreamResult(out);
+            transformer.transform(source, result);
+        } catch (TransformerException e) {
+            LOG.fatal("Failed to transform " + invoiceFile.getName() + " xml." + e);
+            throw new RuntimeException(e);
+        }
+
+        LOG.info("addNamespaceDefinition() Namespace validation completed");
+
+        return out.toByteArray();
+    }
+
+    protected void removeUnsupportedXmlnsAttributes(Element element) {
+        NamedNodeMap attributesMap = element.getAttributes();
+        Attr[] attributes = new Attr[attributesMap.getLength()];
+        
+        for (int i = 0; i < attributesMap.getLength(); i++) {
+            attributes[i] = (Attr) attributesMap.item(i);
+        }
+        
+        Arrays.stream(attributes)
+                .filter(this::isUnsupportedXmlnsAttribute)
+                .forEach(element::removeAttributeNode);
+    }
+
+    protected boolean isUnsupportedXmlnsAttribute(Attr attribute) {
+        return StringUtils.startsWith(attribute.getName(), XMLNS_ATTRIBUTE_PREFIX)
```

Comment:
should this be starts with ignore case?

Reply:
Good catch, I will update this to be case-insensitive.